Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceBreakpointHandler.java

```diff
@@ -46,15 +40,8 @@ public void registerBreakpoint(@NotNull final XLineBreakpoint<XBreakpointPropert
   public void unregisterBreakpoint(@NotNull final XLineBreakpoint<XBreakpointProperties> xBreakpoint, boolean temporary) {
     myXBreakpoints.remove(xBreakpoint);
 
-    final Collection<Pair<String, String>> isolateAndVmBreakpointIds = myXBreakpointToIsolateAndVmBreakpointIdsMap.remove(xBreakpoint);
```

Comment:
Good catch, and I can confirm that that's what was happening. I added a line in `IsolateBreakpointInfo.unregisterBreakpoint()` to clear the map entry for the given xBreakpoint.